### PR TITLE
Functions: added Location.DONHosted

### PR DESCRIPTION
**Pull request by pinebit**

### CODE REVIEW

**Comment by KuphJr** [comment]
Should we add a `addDonSecrets` function to Functions.sol similar to `addRemoteSecrets` to make it easier to construct test requests? https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/functions/dev/0_0_0/Functions.sol

**Comment by KuphJr** [comment]
Also, you may want to ping @justinkaseman for help with getting this PR through the CI workflow.  There may be some gas costs that need to be updated somewhere, but I can't remember.

File: contracts/src/v0.8/functions/dev/0_0_0/Functions.sol [hunk 1]
```diff
@@ -13,7 +13,8 @@ library Functions {
 
   enum Location {
     Inline,
-    Remote
+    Remote,
+    DON
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/0_0_0/Functions.sol:17]: I'm wondering if "DonHosted" or "DONHosted" is a better name?

"secrets are inline"
"secrets are remote"
"secrets are DON-hosted"

File: core/services/ocr2/plugins/functions/plugin.go [hunk 2]
```diff
@@ -104,6 +104,7 @@ func NewFunctionsServices(functionsOracleArgs, thresholdOracleArgs, s4OracleArgs
 		conf.Job,
 		bridgeAccessor,
 		pluginORM,
+		s4ORM,
```
Comment (bolekk) [inline core/services/ocr2/plugins/functions/plugin.go:107]: Why not pass the same Storage object that GatewayConnector receives, instead of the ORM?

**Review by bolekk — CHANGES_REQUESTED** [review]

File: core/services/ocr2/plugins/functions/plugin.go (hunk 2 above)
Comment (pinebit) [reply core/services/ocr2/plugins/functions/plugin.go:107]: Yeah I think this is right, will change

Comment (pinebit) [reply core/services/ocr2/plugins/functions/plugin.go]: Updated.

**Review by bolekk — APPROVED** [review]

**Pull request merged**